### DISCUSSION TITLE
Publish-Item output

### DISCUSSION THREAD

**Discussion by simoncroak** [SitecorePowerShell/Console#1258] [Q&A]
Hi Everyone,
I'm using SPE to help automate our release process. One of the last pieces I need to automate is a full site publish.

I know I can do this using the following command:
`Publish-Item -Path master:\sitecore -Recurse -PublishMode Smart -Target web` 

However there is no output to that command, so I'm unable to tell when it completes,
I notice that the following output is written to the Publishing log once the publish is complete:

`ManagedPoolThread #1 09:26:53 INFO  Starting [Publishing] - AddItemsToQueue`
`ManagedPoolThread #1 09:26:53 INFO  Finished [Publishing] - AddItemsToQueue in 0 ms`
`ManagedPoolThread #1 09:26:53 INFO  Starting [Publishing] - ProcessQueue`
`ManagedPoolThread #1 09:37:57 INFO  Finished [Publishing] - ProcessQueue in 663573 ms`
`ManagedPoolThread #1 09:37:57 INFO  Publish Mode : Smart`
`ManagedPoolThread #1 09:37:57 INFO  Created : 383`
`ManagedPoolThread #1 09:37:57 INFO  Updated : 427`
`ManagedPoolThread #1 09:37:57 INFO  Deleted : 0`
`ManagedPoolThread #1 09:37:57 INFO  Skipped : 190670`

Any ideas how I get this sort of output to the SPE console? Instead of the log on the server?

Thanks

Simon

**Comment by michaellwest** [comment]
The PowerShell command is simply calling the back end API. I don't think there is much we can do for the ootb command. You could probably get a reference to the job and see if the process counter goes up. Then you would have a while loop with a sleep command.
